docs/latest/design/request-authentication.md

```diff
@@ -0,0 +1,537 @@
+dynamic_active_clusters:
+  - cluster:
+      name: default-backend-rule-0-match-0-www.example.com
+      ...
+        endpoints:
+        - locality: {}
+          lb_endpoints:
+            - endpoint:
+                address:
+                  socket_address:
+                    address: $BACKEND_SERVICE1_IP
+                    port_value: 3000
+  - cluster:
+      '@type': type.googleapis.com/envoy.config.cluster.v3.Cluster
+      name: default-backend-rule-1-match-0-www.example.com
+      ...
+        endpoints:
+        - locality: {}
+          lb_endpoints:
+            - endpoint:
+                address:
+                  socket_address:
+                    address: $BACKEND_SERVICE2_IP
+                    port_value: 3000
+...
+```
+
+__Note:__ The JWT provider cluster and route is omitted from the above example for brevity.
+
+### Implementation Outline
+
+* Create a `security` API group and add the Authentication API type to this group.
+* Update the Kubernetes provider to get/watch Authentication resources that are referenced by managed HTTPRoutes. Add
+  the referenced Authentication object to the resource map and publish it.
+* Update the resource translator to include the Authentication API in HTTPRoute processing.
+* Update the xDS translator to translate an Authentication into xDS resources. The translator should perform the
+  following:
+  * Convert a list of JWT rules from the xds IR into an Envoy JWT filter config.
+  * Create a JWT authentication filter.
+  * Build the HTTP Connection Manager (HCM) HTTP filters.
+  * Build the HCM.
+  * When building the Listener, create an HCM for each filter-chain.
+
+## Adding Authentication Provider Types
+
+Additional authentication provider types can be added in the future through the `ProviderType` API. For example, to add
+the `Foo` authentication provider type:
+
+Define the `FooProvider` type:
+
+```go
+package v1alpha1
+
+// FooProvider defines the "Foo" authentication provider type.
+type FooProvider struct {
+	// TODO: Define fields of the Foo authentication provider type.
+}
+```
+
+Add the `FooProvider` type to `AuthenticationSpec`:
+
+```go
+package v1alpha1
+
+type AuthenticationSpec struct {
+	...
+	
+	// Foo defines the Foo authentication type. For additional
+	// details, see:
+	//
+	//   <INSERT_LINK>
+	//
+	// +optional
+	Foo *FooAuthentication `json:"foo"`
+}
+```
+
+Authentication should support additional authentication types in the future, for example:
+- mutualTLS (client certificate)
+- OAuth2
+- OIDC
+- External authentication
+
+## Outstanding Questions
+
+- If Envoy Gateway owns the Authentication API, is an xDS IR equivalent needed?
+- Should local JWKS be implemented before remote JWKS?
+- How should Envoy obtain the trusted CA for a remote JWKS?
+- Should HTTPS be the only supported scheme for remote JWKS?
+- Should OR'ing JWT providers be supported?
+- Should Authentication provide status?
```

Comment:
would be good to add, but imho doesnt need to be added right away

Reply:
Definitely worth considering what status updates would make sense.

Reply:
Responded here: https://github.com/envoyproxy/gateway/pull/529#issuecomment-1332825708